lib/grp.gi

```diff
@@ -939,6 +939,64 @@ InstallMethod( CompositionSeries,
     "for simple group", true, [IsGroup and IsSimpleGroup], 100,
     S->[S,TrivialSubgroup(S)]);
 
+InstallMethod(CompositionSeriesThrough,"intersection/union",IsElmsColls,
+  [IsGroup and IsFinite,IsList],0,
+function(G,normals)
+local cs,i,j,pre,post,c,new,rev;
+  cs:=CompositionSeries(G);
+  # find normal subgroups not yet in
+  normals:=Filtered(normals,x->not x in cs);
+  # do we satisfy by sheer dumb luck?
+  if Length(normals)=0 then return cs;fi;
+
+  SortBy(normals,x->-Size(x));
+  # check that this is a valid series
+  Assert(0,ForAll([2..Length(normals)],i->IsSubset(normals[i-1],normals[i])));
```

Comment:
Just out of curiosity, is there a specific reason to use `IsSubset` instead of `IsSubgroup` here?

Reply:
Perhaps there should also be a second assertion (perhaps run only at a higher assertion level) which verifies that `normals` truly consists of normal subgroups; and/or add a warning to the documentation that passing non-normal subgroups may result in wrong output?